WordPress/WordPressTest/Dashboard/Dynamic Cards/BlogDashboardDynamicCardCoordinatorTests.swift

```diff
@@ -65,9 +65,95 @@ final class BlogDashboardDynamicCardCoordinatorTests: XCTestCase {
         }))
     }
 
+    func test_didTapCardInvokesLinkRouterWhenURLMatches() {
+        // Given
+        let (id, url) = ("123", "https://wordpress.com")
+        var mockRouter = MockRouter(routes: [])
+        let expectation = expectation(description: "LinkRouter must invoke completion")
+        var isLinkInvoked = false
+
+        mockRouter.completion = { _, _ in
+            isLinkInvoked = true
+            expectation.fulfill()
+        }
+        let coordinator = makeCoordinator(id: id, url: url, linkRouter: mockRouter)
+
+        // When
+        coordinator.didTapCard()
+
+        // Then
+        wait(for: [expectation], timeout: 1)
+        XCTAssert(isLinkInvoked)
+    }
+
+    func test_didTapCardDoesNotInvokeLinkRouterWhenURLInvalid() {
+        // Given
+        let (id, url) = ("123", "gibberish")
+        var mockRouter = MockRouter(routes: [])
+        mockRouter.canHandle = false
+        var isLinkInvoked = false
+
+        mockRouter.completion = { _, _ in
+            isLinkInvoked = true
+            XCTFail("completion shouldn't have ran")
+        }
+        let coordinator = makeCoordinator(id: id, url: url, linkRouter: mockRouter)
+
+        // When
+        coordinator.didTapCard()
+
+        // Then
+        XCTAssertFalse(isLinkInvoked)
+    }
+
+    func test_didTapCardCTAInvokesLinkRouterWhenURLMatches() {
+        // Given
+        let (id, url) = ("123", "https://wordpress.com")
+        var mockRouter = MockRouter(routes: [])
+        let expectation = expectation(description: "LinkRouter must invoke completion")
+        var isLinkInvoked = false
+
+        mockRouter.completion = { _, _ in
+            isLinkInvoked = true
+            expectation.fulfill()
+        }
+        let coordinator = makeCoordinator(id: id, url: url, linkRouter: mockRouter)
+
+        // When
+        coordinator.didTapCardCTA()
+
+        // Then
+        wait(for: [expectation], timeout: 1)
+        XCTAssert(isLinkInvoked)
+    }
+
+    func test_didTapCardCTADoesNotInvokeLinkRouterWhenURLInvalid() {
+        // Given
+        let (id, url) = ("123", "gibberish")
+        var mockRouter = MockRouter(routes: [])
+        mockRouter.canHandle = false
+        var isLinkInvoked = false
+
+        mockRouter.completion = { _, _ in
+            isLinkInvoked = true
+            XCTFail("completion shouldn't have ran")
+        }
+        let coordinator = makeCoordinator(id: id, url: url, linkRouter: mockRouter)
+
+        // When
+        coordinator.didTapCardCTA()
+
+        // Then
+        XCTAssertFalse(isLinkInvoked)
+    }
```

Comment:
Suggestion: In my opinion, this test should check whether the web view is shown instead of checking whether the link router is called or not. Here is one way to achieve that:
1. Create a `UIViewController` instance in the test method.
2. Pass that VC instance to `makeCoordinator`
3. Call `coordinator.didTapCard()`
4. Assert `viewController.presentedViewController` type is `WebKitViewController`. And you can also assert the web view url equals `model.payload.url`

Let me know what you think.

Reply:
I think that indicates, this test should really be a UI test. Because the output is a change in the view. I don't really like testing for function calls too. But the change in output occurs on UIKit. I am not against it. Though I don't think it is the right pattern to apply in general.

Reply:
> I don't really like testing for function calls too.

This is find in the case where URL is valid. My suggestion is to test 2 cases:

1. Unit test to check the link router is called when the URL is Valid. This is done ✅ 
2. Unit test to check a web view is shown the URL is not valid.

These 2 cases are enough to cover the whole `routeToDestination` method.

Also, I think it's okey to Unit Test UI, it's much simpler and faster than writing UITests.